assets/js/blocks/reviews-by-product/frontend-block.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { getOrderArgs, getReviews } from './utils';
+import LoadMoreButton from '../../frontend/components/load-more-button';
+import OrderBySelect from '../../frontend/components/order-by-select';
+import Reviews from '../../frontend/components/reviews';
+import withComponentId from '../../frontend/hocs/with-component-id';
+
+/**
+ * Block rendered in the frontend.
+ */
+class FrontendBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		const { attributes } = this.props;
+
+		this.state = {
+			orderby: attributes.orderby,
+			reviews: [],
+			totalReviews: 0,
+		};
+
+		this.onChangeOrderby = this.onChangeOrderby.bind( this );
+		this.appendReviews = this.appendReviews.bind( this );
+	}
+
+	componentDidMount() {
+		this.loadFirstReviews();
+	}
+
+	getDefaultArgs() {
+		const { attributes } = this.props;
+		const { order, orderby } = getOrderArgs( this.state.orderby );
+		const { productId, reviewsOnPageLoad } = attributes;
+
+		return {
+			order,
+			orderby,
+			per_page: reviewsOnPageLoad,
+			product_id: productId,
+		};
+	}
+
+	loadFirstReviews() {
+		const args = this.getDefaultArgs();
+
+		getReviews( args ).then( ( { reviews, totalReviews } ) => {
+			this.setState( { reviews, totalReviews } );
+		} ).catch( () => {
+			this.setState( { reviews: [] } );
+		} );
+	}
+
+	appendReviews() {
+		const { attributes } = this.props;
```

Comment:
This logic seems like something that could be extracted as a general "LoadMore.." type component. Not necessary here, but just something to think about.  I could see a `withLoadMore` hoc that basically enhances it's wrapped component with a `LoadMore` button wired up to necessary logic for performing the additional items.

I know there's some things to work out how it would work (where would state live?) but if you think like this would be a nice to have considering all the potential future blocks that are using this, it might be a good thing for me to tackle in a separate pull?

Reply:
That's a good idea, I think it would be interesting to implement this at the same time we implement the other reviews blocks (_Reviews by Category_ and _All Reviews_) so not blocking this PR.